shipyard-controller/handler/sequencedispatcher.go

```diff
@@ -147,6 +146,61 @@ func (sd *SequenceDispatcher) dispatchSequences() {
 	}
 }
 
+func (sd *SequenceDispatcher) isSequenceBlocked(queueItem models.QueueItem) (bool, error) {
+	// searching for running sequences
+	startedSequenceExecutions, err := sd.sequenceExecutionRepo.Get(models.SequenceExecutionFilter{
+		Scope: models.EventScope{
+			EventData: keptnv2.EventData{
+				Project: queueItem.Scope.Project,
+				Stage:   queueItem.Scope.Stage,
+				Service: queueItem.Scope.Service,
+			},
+		},
+		Status: []string{apimodels.SequenceStartedState},
+	})
+	if err != nil {
+		log.Errorf("Could not load started sequences for project %s, service %s, stage %s: %v", queueItem.Scope.Project, queueItem.Scope.Service, queueItem.Scope.Stage, err)
+		return true, err
+	}
+
+	if len(startedSequenceExecutions) > 0 {
+		log.Infof("Sequence with KeptnContext %s blocked due to started sequence with KeptnContext %s in stage %s", queueItem.Scope.KeptnContext, startedSequenceExecutions[0].Scope.KeptnContext, queueItem.Scope.Stage)
+		return true, nil
+	}
+
+	//searching for triggered sequences which were triggered before the actual sequence
+	triggeredSequenceExecutions, err := sd.sequenceExecutionRepo.Get(models.SequenceExecutionFilter{
+		Scope: models.EventScope{
+			EventData: keptnv2.EventData{
+				Project: queueItem.Scope.Project,
+				Stage:   queueItem.Scope.Stage,
+				Service: queueItem.Scope.Service,
+			},
+		},
+		Status:      []string{apimodels.SequenceTriggeredState},
+		TriggeredAt: queueItem.Timestamp,
+	})
+	if err != nil {
+		log.Errorf("Could not load triggered sequences for project %s, service %s, stage %s: %v", queueItem.Scope.Project, queueItem.Scope.Service, queueItem.Scope.Stage, err)
+		return true, err
+	}
+
+	if len(triggeredSequenceExecutions) == 1 {
+		if triggeredSequenceExecutions[0].Scope.KeptnContext != queueItem.Scope.KeptnContext {
+			log.Infof("Sequence with KeptnContext %s is blocked due to triggered sequence with KeptnContext %s in stage %s", queueItem.Scope.KeptnContext, triggeredSequenceExecutions[0].Scope.KeptnContext, queueItem.Scope.Stage)
+			return true, nil
+		}
+	}
+
+	if len(triggeredSequenceExecutions) > 1 {
+		log.Infof("Sequence with KeptnContext %s is blocked due to triggered sequences in stage %s", queueItem.Scope.KeptnContext, queueItem.Scope.Stage)
+		return true, nil
+	}
+
+	log.Infof("Sequence with KeptnContext %s is not blocked by any other sequence in stage %s", queueItem.Scope.KeptnContext, queueItem.Scope.Stage)
```

Comment:
i think we can remove this log line, as this will otherwise make the logs very verbose

Reply:
adapted